src/runtime/SrcDocIFrameController.ts

```diff
@@ -53,13 +70,30 @@ export class SrcDocIFrameController extends Disposable implements IEmbeddedConte
 
         await this.setRootClassNames(this.config.classNames);
         await this.setRootStyles(this.config.styles);
+        await this.addAnchorLinkHandlers();
 
         this.updateIFrameHeight(
             // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
             this.host.contentWindow!.document.body.getBoundingClientRect().height,
         );
     }
 
+    // finds all relative links (href^="#") and changes their click behavior
+    addAnchorLinkHandlers() {
+        const document = this.host.contentWindow?.document;
```

Comment:
I think that for consistency's sake it would be better to use non-null assertion `!` here as well, since I couldn't verify anywhere on MDN that `contentWindow` can actually be `null`.